oak_functions/client/java/src/com/google/oak/functions/client/AttestationClient.java

```diff
@@ -72,42 +72,32 @@ public VerificationException(String msg) {
     }
   }
 
-  /** Creates an unattested AttestationClient instance. */
-  public AttestationClient() {}
-
-  // TODO(#2356): Change the return type to `AttestationResult` instead of throwing exceptions.
   /**
-   * Creates a gRPC channel and creates an attested channel over it.
+   * Adds a gRPC channel interceptor that inserts an API key header into every request.
    *
-   * @param url must contain a protocol used for the connection ("https://" or "http://").
+   * @param builder an instance of a {@code ManagedChannelBuilder} for a gRPC channel.
    * @param apiKey value of the API key used in gRPC requests. If the value is `null` or empty, then
    * the API key header is not included in requests.
    * https://cloud.google.com/docs/authentication/api-keys
-   * @param verifier Checks that the ServerIdentity contains the expected attestation info as
-   * described in {@code ServerIdentityVerifier::verifyAttestationInfo}.
+   *
+   * @return an updated instance of a {@code ManagedChannelBuilder}.
    */
-  public void attest(String url, String apiKey, Predicate<ConfigurationInfo> verifier)
-      throws GeneralSecurityException, IOException, InterruptedException, VerificationException {
-    // Create gRPC channel.
-    URL parsedUrl = new URL(url);
+  public static ManagedChannelBuilder addApiKey(ManagedChannelBuilder builder, String apiKey) {
```

Comment:
I still think this method has nothing to do with attestation and should not be here, also I'm not sure it's idiomatic to get a channel builder and modify it in place from a method like this. In fact, there is probably some utility that does it already which we probably haven't found yet. It's fine to fix later, but do you agree that this should not be here eventually?

Reply:
Yes, we can create a utility library for creating gRPC channels.
Because otherwise all of this code will have to be in the client.
I.e. we are trying to encapsulate the gRPC channel creation logic and at the same time give clients the ability to fully manage the gRPC channel creation.

Filed an issue: https://github.com/project-oak/oak/issues/2411